pkg/apis/application/v1alpha1/types.go

```diff
@@ -437,6 +441,83 @@ type SyncPolicyAutomated struct {
 	Prune bool `json:"prune,omitempty" protobuf:"bytes,1,opt,name=prune"`
 	// SelfHeal enables auto-syncing if  (default: false)
 	SelfHeal bool `json:"selfHeal,omitempty" protobuf:"bytes,2,opt,name=selfHeal"`
+	// MaintenanceWindows disables auto-syncing during the configured time windows
+	MaintenanceWindows MaintenanceWindows `json:"maintenanceWindows,omitempty" protobuf:"bytes,3,opt,name=maintenanceWindows"`
+}
+
+type MaintenanceWindows []*MaintenanceWindow
+
+// MaintenanceWindow controls when automated syncs should be disabled
+type MaintenanceWindow struct {
+	// Schedule is the time the window will begin, specified in cron format
+	Schedule *string `json:"schedule,omitempty" protobuf:"bytes,1,opt,name=schedule"`
+	// Duration is the amount of time the maintenance window will be open
+	Duration *string `json:"duration,omitempty" protobuf:"bytes,2,opt,name=duration"`
+}
+
+func (m *MaintenanceWindows) Active() (bool, error) {
+	if m != nil {
+		specParser := cron.NewParser(cron.Minute | cron.Hour | cron.Dom | cron.Month | cron.Dow)
+		for _, w := range *m {
+			schedule, err := specParser.Parse(*w.Schedule)
+			if err != nil {
+				return false, err
```

Comment:
do we want to try and catch these errors or creation?

Reply:
Yes, that is a good idea. I have implemented it.